### PR TITLE
Added check for Don't Cares in Combinational Analysis Tool

### LINKED ISSUE #659: Popup if Don't Cares present in Combinational Analysis Table
**Is your feature request related to a problem? Please describe.**
While using the Combinational Analysis feature, we've to fill up a table of inputs which can be a lot sometimes. And if we forget to change even a single "x" we don't get a circuit. So the whole work of filling in the table goes to waste due to s small mistake.

**Describe the solution you'd like**
A popup which would show that "x" still exist. Should work till there are no "x" left in the table.

**Are you working on this? (Yes/No)**
Nope. For GCI 2019

### PR DESCRIPTION
**Pull request by sal2701**
Fixes #659 

#### Describe the changes you have made in this pr -

I have checked for Don't Cares in the table and alerted the user if there are any present.

### Screenshots of the changes (If any) -
dontcare

### CODE REVIEW

File: public/js/combinationalAnalysis.js [hunk 1]
```diff
@@ -86,9 +85,25 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
...
             //passing the hash values to avoid spaces being passed which is causing a problem
+
+            //checking for Dont Care's in the table
+            var a = 13; //key where the values are stored
+            var flag = 0; //to indicate error
+            while(!flag){
+                flag = 1;
+                for(i=0; i<Object.keys(data).length; i++) {
+                    if(data[a]['x'].length != 0 && flag) { //to see if there are any don't cares
+                        console.log("Error: Dont cares present");
```
[the remark this answers is not in the document]
Comment (sal2701) [reply public/js/combinationalAnalysis.js]: done

File: public/js/combinationalAnalysis.js [hunk 2]
```diff
@@ -86,9 +85,24 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
...
           text: "Generate Circuit",
           click: function() {
-            $( this ).dialog( "close" );
             var data = generateBooleanTableData(outputListNamesInteger);
             //passing the hash values to avoid spaces being passed which is causing a problem
+
+            //checking for Dont Care's in the table
+            var a = 13; //key where the values are stored
+            var flag = 0; //to indicate error
+            while(!flag) {
```
[the remark this answers is not in the document]
Comment (sal2701) [reply public/js/combinationalAnalysis.js]: done

File: public/js/combinationalAnalysis.js [hunk 3]
```diff
@@ -86,9 +85,25 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
...
+
+            //checking for Dont Care's in the table
+            var a = 13; //key where the values are stored
+            var flag = 0; //to indicate error
+            while(!flag){
+                flag = 1;
+                for(i=0; i<Object.keys(data).length; i++) {
+                    if(data[a]['x'].length != 0 && flag) { //to see if there are any don't cares
+                        console.log("Error: Dont cares present");
+                        alert("Don't leave any X's in the table");
```
[the remark this answers is not in the document]
Comment (sal2701) [reply public/js/combinationalAnalysis.js]: It is needed in this case

**Comment by satu0king** [comment]
Closing this PR due to https://github.com/CircuitVerse/CircuitVerse/issues/659#issuecomment-602441891

**Pull request closed without merging**